src/Bicep.Core/Modules/ModuleReference.cs

```diff
@@ -13,6 +15,8 @@ protected ModuleReference(string scheme)
             this.Scheme = scheme;
         }
 
+        protected static Regex ModuleAliasNameRegex { get; } = new(@"[\w-]");
```

Comment:
>new [](http://example.com/codeflow?start=63&length=3)

Does this need a `Compile` flag or any others?

Reply:
Good catch. Added `Compile` and `CultureInvariant` flags. Also changed `\w` to `[a-zA-Z0-9-_]`.